src/token/libexec/token/token-permit

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+### token-permit -- sign (and submit) EIP2612 or DAI style permits
+set -e
+display_usage() {
+  echo "Usage: token permit <token> <spender> [<amount|allowed>] [<deadline|expiry>] [--submit]"
+}
+
+for i in "$@"; do
+  case $i in
+    --submit)
+      SUBMIT=yes
+      shift
+      ;;
+    *)
+      shift
+      set -- "$@" "$i"
+      ;;
+  esac
+done
+
+if [ $# -lt 2 ];  then
+    display_usage
+    exit 1
+fi
+
+TOKEN=$1
+SPENDER=$2
+
+EIP2612_TYPEHASH=$(seth keccak $(seth --from-ascii "Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"))
+PERMIT_TYPEHASH=$(seth call $TOKEN "PERMIT_TYPEHASH()(bytes32)")
```

Comment:
what will `PERMIT_TYPEHASH` evaluate too if this call fails (i.e. there is no `PERMIT_TYPEHASH()` method on the contract)?

Reply:
I don't know but the update to EIP2616 is making it part of the standard that it should be a public method so we should be able to rely on it evaluating properly. I'll check

Reply:
Actually, the PERMIT_TYPEHASH is not added to 2612, but DOMAIN_SEPARATOR is.
But this feels okay to me anyway for the purposes of this script